src/main.ts

```diff
@@ -30,14 +31,24 @@ export async function run() {
       let token = core.getInput('token');
       let auth = !token ? undefined : `token ${token}`;
 
+      const manifest = await installer.getManifest(auth);
+
       const checkLatest = core.getBooleanInput('check-latest');
+
       const installDir = await installer.getGo(
         versionSpec,
         checkLatest,
         auth,
-        arch
+        arch,
+        manifest
       );
 
+      if (IS_WINDOWS) {
+        versionSpec = installDir.split('\\').reverse()[1];
+      } else {
+        versionSpec = installDir.split('/').reverse()[1];
+      }
+
```

Comment:
Could you please change it to `const version = path.basename(path.dirname(installDir))`